demo-app/src/commonMain/kotlin/org/maplibre/compose/demoapp/demos/UserLocationDemo.kt

```diff
@@ -74,11 +75,12 @@ object UserLocationDemo : Demo {
 
     LocationPuck(
       idPrefix = "user-location",
-      locationState = state.locationState,
+      location = state.locationState.location,
       cameraState = state.cameraState,
       accuracyThreshold = 0f,
-      showBearing = bearingUpdate != BearingUpdate.IGNORE,
-      showBearingAccuracy = bearingUpdate != BearingUpdate.IGNORE,
+      bearing =
+        if (bearingUpdate != BearingUpdate.IGNORE) state.locationState.location?.orientation
+        else null,
```

Comment:
```suggestion
      bearing = state.locationState.location?.orientation?.takeUnless { bearingUpdate == BearingUpdate.IGNORE }
```

IMHO this is one of the few use cases for `takeUnless`